### PR TITLE
Add more metrics around connection buffers

### PR DESCRIPTION
**Pull request by kegsay**

### CODE REVIEW

File: sync3/connmap.go [hunk 1]
```diff
@@ -27,17 +33,61 @@ func NewConnMap() *ConnMap {
...
+			Subsystem: "api",
+			Name:      "expiry_conn_buffer_full",
+			Help:      "Counter of expired API connections due to reaching buffer update limit",
+		})
+		prometheus.MustRegister(cm.expiryBufferFullCounter)
+		cm.numConns = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "sliding_sync",
+			Subsystem: "api",
+			Name:      "num_active_conns",
+			Help:      "Number of active sliding sync connections.",
```
Comment (DMRobertson) [inline sync3/connmap.go:56]: Doesn't this metric already exist?

Comment (DMRobertson) [reply sync3/connmap.go:56]: Oh, you moved it from the SyncLiveHandler.

File: sync3/connmap.go [hunk 2]
```diff
@@ -27,17 +33,61 @@ func NewConnMap() *ConnMap {
...
-func (m *ConnMap) Len() int {
+// UpdateMetrics recalculates the number of active connections. Do this when you think there is a change.
+func (m *ConnMap) UpdateMetrics() {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	return len(m.connIDToConn)
+	m.updateMetrics(len(m.connIDToConn))
+}
+
+// updateMetrics is like UpdateMetrics but doesn't touch connIDToConn and hence need a lock. We use this internally
```
Comment (DMRobertson) [inline sync3/connmap.go:84]: should this be "and hence doesn't need a lock?"

**Comment by kegsay** [comment]
Yes I sanity checked with SYNCV3_PROM.

**Pull request merged**